[13.0][IMP] website_sale_product_minimal_price: Forward port of improvements of 12.0

CarlosRoca13 opened the pull request:
Forward port adapting the minimal price computing to the new way added on version 13.0.

With the changes added the price takes care on the website selected pricelist and it's added the grid of price scale for each product.variant.

cc @Tecnativa TT24175

please @sergio-teruel @pedrobaeza review this.

CarlosRoca13 committed:
[IMP] website_sale_product_minimal_price: Forward port of improvements of 12.0

Forward port adapting the minimal price computing to the new way added on version 13.0.

With the changes added the price takes care on the website selected pricelist and it's added the grid of price scale for each product.variant.

TT24175

pedrobaeza commented on website_sale_product_minimal_price/__manifest__.py:
```diff
@@ -12,7 +12,7 @@
     "application": False,
     "installable": True,
-    "depends": ["website_sale"],
+    "depends": ["website_sale_stock"],
```
Why changing this dependency?

CarlosRoca13 replied:
Because the module that adds the method `VariantMixin._onChangeCombinationStock` is this other

pedrobaeza replied:
And can we avoid that one and use one in pure `website_sale` module? It's not logical to require stock for this. What about if you sell services?

CarlosRoca13 replied:
Ok I will try to use the website_sale module

pedrobaeza commented on website_sale_product_minimal_price/controllers/main.py:
```diff
@@ -19,30 +19,74 @@ def get_combination_info_minimal_price(self, product_template_ids, **kw):
+                    ("min_quantity", ">", 0),
+                ]
+            )
+            .mapped("min_quantity")
```
Do this in 2 steps for better black formatting: first obtain items records, and then apply `mapped`.

CarlosRoca13 commented:
@pedrobaeza change done

pedrobaeza commented on website_sale_product_minimal_price/controllers/main.py:
```diff
@@ -19,30 +19,70 @@ def get_combination_info_minimal_price(self, product_template_ids, **kw):
+        # price scale.
+        qty_list = request.env["product.pricelist.item"].search(
+            [
+                ("|"),
```
I'm seeing that this is incorrect syntax. It should be just:
```suggestion
                "|",
```

If it works, it's by chance.

CarlosRoca13 commented:
All changes done @pedrobaeza

chienandalu reviewed (commented):
Can you point to the fowarded commits?

CarlosRoca13 commented:
@chienandalu yes sure, here you have them:
- https://github.com/OCA/e-commerce/commit/35f35142060413965523bd75f757798fecbbd368#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/bac82bf23b59241d097ae37b59873bbb6fe5e3f3#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/d81b0075f1f242b4bb267e0bc21a8e6b5f1b7ce4#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/3964915256789e326902dd4fbcc40eb08bfade71#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/ac744844b441cb6ac6fd9e6ce3fd82e142359f3f#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe

And It's added the new grid price compute from this PR: https://github.com/OCA/e-commerce/pull/539

CarlosRoca13 committed:
[FIX] website_sale_product_minimal_price: Take into account all pricelist item

When the product has a sales price = 0, his variants has no extra price and a pricelist depends on other pricelists to compute the prices, the algorithm takes a random variant.

Doing this changes we take care all pricelists that depends on the main to compute the prices asociated to the product.

This changes are forward ported and addapted to 13.0 version from this commit https://github.com/OCA/e-commerce/commit/fe99a35ae3a49bce24362c8b4e14aa152e2ae210#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe

CarlosRoca13 commented:
Added the changes accepted on this PR https://github.com/OCA/e-commerce/pull/539 doing the necessary changes for allow working with the new version

pedrobaeza commented:
/ocabot merge minor
